src/Middleware/ImplicitHeadMiddleware.php

```diff
@@ -89,7 +89,9 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
         }
 
         $response = $delegate->process(
-            $request->withMethod(RequestMethod::METHOD_GET)
+            $request
+                ->withMethod(RequestMethod::METHOD_GET)
+                ->withAttribute('forwarded_http_method', RequestMethod::METHOD_HEAD)
```

Comment:
Create a constant for this attribute name, please, and then update the code to use that constant.

Reply:
And, on second thought: shouldn't this be accomplished with either an `X-Forwarded-Proto` or `Forwarded` header (see [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto))?

Reply:
This headers should be using in specified cases but not to forward http methods.